marblerun/features/secrets-management.md

```diff
@@ -2,28 +2,25 @@
 
-Crucially, in SGX, the aforementioned `cpu_secret` is hard-wired in silicon and is unique per physical processor. Thus, sealing keys do no commute between processors. Particularly in cloud settings, this becomes a problem: if a VM is scheduled on a different host, enclaves running within that VM cannot unseal their local data anymore, because the host's `cpu_secret` has changed.
+If you use one key throughout multiple Marbles, care has to be taken to not repeat nonces between Marbles when using shared keys with AES-GCM or similar encryption algorithms.
 
-To solve this problem, the Coordinator generates "virtual sealing keys" for Marbles. After successful validation of a Marble, the Coordinator injects a Marble's virtual sealing key through the `{{ hex .Marblerun.SealKey }}` placeholder in the `Parameters` section of the manifest.
+## Certificates
+Upon launch of a Marble, the Coordinator will generate a private key for each new Marble and issue a corresponding X.509 certificate. Both are automatically added to the Marble's environment variables as `MARBLE_PREDEFINED_MARBLE_CERTIFICATE_CHAIN` and `MARBLE_PREDEFINED_PRIVATE_KEY`. In addition, you can choose to define them to other environment variables or copy them to a file only available to the Marble through by defining template placeholders such as `{{ pem .MarbleRun.MarbleCert.Cert }}` or `{{ pem .MarbleRun.MarbleCert.Private }}` in the manifest. The certificate is signed with the Coordinator's intermediate CA. Marbles can use this certificate to establish secure communication channels with other Marbles and external clients (i.e., users of your app). Clients only need to verify the Coordinator's root CA once before they can securely communicate with any Marble, as is described in more detail in our [verification hands-on](workflows/verification.md).
 
-The Coordinator derives virtual sealing keys from respective Marble's ID and a master secret only known to the Coordinator using the [HKDF](https://tools.ietf.org/html/rfc5869) scheme. The ID is a public 128-bit value randomly generated by each Marble for itself upon startup and persisted in plaintext on local storage. Consequently, two Marbles could end up with the same ID and virtual sealing key under certain circumstances. The security implications of this need to be taken into account when using virtual sealing keys. For example, when using virtual sealing keys with AES-GCM, one must only use cryptographically random nonces.
+Next to this automatically generated certificate, you can also define additional ones  as secrets in your manifest. These are also signed by the Coordinator's intermediate CA and generated upon each launch, but do not contain a specific Marble's by default. You can include them via placeholders such as `{{ pem .Secrets.MySecretCertificate }}` in your manifest.
```

Comment:
`but do not contain a specific Marble's by default` 
Is there smth missing?

Reply:
Was supposed to be: `a specific Marble's identity`